lineman/app/components/flash/flash.coffee

```diff
@@ -3,4 +3,24 @@ angular.module('loomioApp').directive 'flash', ->
   restrict: 'E'
   templateUrl: 'generated/components/flash/flash.html'
   replace: true
-  controller: 'FlashController'
+  controllerAs: 'flash'
+  controller: ($scope, $timeout, FlashService) ->
+    @pendingDismiss = null
+
+    $scope.$on 'flashMessage', (event, flash) =>
+      @flash = flash
+      $timeout.cancel @pendingDismiss if @pendingDismiss?
+      @pendingDismiss = $timeout @dismiss, 2000
+
+    @modalIsVisible = ->
+      angular.element('.modal').hasClass('in')
+
+    @display = => @flash and (@modal == @modalIsVisible())
+    @dismiss = => @flash = null
+
+    FlashService.success window.Loomio.flash.success if window.Loomio.flash.success?
```

Comment:
Maybe this init block should be moved to app.coffee in the future? Anyway this whole thing looks great. Thanks @gdpelican.